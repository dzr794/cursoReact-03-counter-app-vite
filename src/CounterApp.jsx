import PropTypes from 'prop-types'
import { useState } from 'react'

export const CounterApp = ( {value} ) => {

  const [ counter, setCounter ] = useState( value );

  const handleSubtract = () => {
    setCounter( counter - 1 );
  }

  const handleAdd = () => {
    setCounter( counter + 1 );
    // setCounter( (c) => c + 1 );
  }

  const handleReset = () => {
    setCounter( value );
  }

  return (
    <>
    <h1>CounterApp</h1>
    <h2 data-testid="counter-value" > { counter } </h2>
    <button data-testid="substract-1" onClick={handleSubtract}>-1</button>
    <button data-testid="add-1" onClick={handleAdd}>+1</button>
    <button data-testid="reset" onClick={handleReset}>Reset</button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}

CounterApp.defaultProps = {
  value: 0,
}