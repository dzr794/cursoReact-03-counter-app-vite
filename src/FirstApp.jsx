import PropTypes from 'prop-types'

export const FirstApp = ( { title, subtitle, name } ) => {
  return (
    <>
    <h1> {title} </h1>
    <p> {subtitle} </p>
    <p> {name} </p>
    </>
  )
}

FirstApp.propTypes = {
  name: PropTypes.string,
  subtitle: PropTypes.string,
  title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
  name: "David",
  subtitle: "No hay subtitulo",
  title: "No hay t√≠tulo",
}