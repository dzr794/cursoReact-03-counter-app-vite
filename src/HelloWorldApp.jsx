// export function App() {
//   return (<h1>Holiwa !!!</h1>);
// }

// * usando el snipet "rafc"

const myFunction = (a,b) => a+b;

export const HelloWorldApp = () => {
  return (
    <h1>{myFunction(1,2)}</h1>
  )
}
