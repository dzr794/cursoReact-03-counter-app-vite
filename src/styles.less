@font-color: #f1f1f1;

html{
  body{
    background-color: #212121;
    color: @font-color;
    font-family: Helvetica, Arial, sans-serif;
    padding: 30px;
  }
}