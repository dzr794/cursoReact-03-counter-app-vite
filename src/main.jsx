import React from "react";
import ReactDOM from "react-dom/client"
// import { HelloWorldApp } from "./HelloWorldApp";
// import { FirstApp } from "./FirstApp";
import { CounterApp } from "./CounterApp";

import './styles.less';

ReactDOM.createRoot( document.getElementById('root') ).render(
  <React.StrictMode>
    {/* <FirstApp title="Holi"/> */}
    <CounterApp value={20} />
  </React.StrictMode>
);