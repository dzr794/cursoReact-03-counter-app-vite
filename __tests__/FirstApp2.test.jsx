import {render, screen} from '@testing-library/react'
import {FirstApp} from '../src/FirstApp'

describe('Pruebas en <FirstApp />', () => { 
  
  const title = 'Holi soy el tÃ­tulo';
  const subTitle = 'Soy el subtitulo';

  test('Debe hacer match con el snapshot', () => { 
    const {container} = render( 
      <FirstApp 
        title={ title }
      />
    );

    expect(container).toMatchSnapshot();
  })

  test('El titulo debe ser un H1', () => { 
    render( 
      <FirstApp 
        title={ title }
      />
    );

    expect( screen.getByRole('heading', { level:1 }).innerHTML ).toContain( title );
  });

  test('Deben haber 2 subrtitulos', () => {
    render( 
      <FirstApp 
        title={ title }
        subtitle={ subTitle }
      />
    );

    expect( screen.getAllByText(subTitle).length ).toBe(2);
  });

});