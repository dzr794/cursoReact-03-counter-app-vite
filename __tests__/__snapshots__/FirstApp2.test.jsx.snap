// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Holi soy el t√≠tulo
     
  </h1>
  <p>
     
    No hay subtitulo
     
  </p>
  <p>
     
    No hay subtitulo
     
  </p>
  <p>
     
    David
     
  </p>
</div>
`;
